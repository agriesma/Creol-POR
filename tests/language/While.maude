load creol-interpreter
mod PROGRAM is
protecting CREOL-SIMULATOR .
op init : -> Configuration [ctor] .
eq init =
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds: 
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return ( emp ) > *
  < "run" : Mtdname | Param: noVid, Latt: "l" |-> null , "f" |-> null, Code: while "/=" ( "l" # list(emp) ) do  ( "f", "l" )  ::= "head" ( "l" ) # "tail" ( "l" ) od  ; return ( emp ) >, Ocnt: 0 >

.
endm
