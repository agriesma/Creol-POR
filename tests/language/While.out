==========================================
reduce in PROGRAM : init .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noVid, Att: noSubst, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: skip ; return(emp) >
    *
  < "run" : Mtdname | Param: noVid, Latt: "f" |-> null, "l" |-> null, Code:
    while "/="("l" :: list(emp)) do ("f", "l") ::= "head"("l") :: "tail"("l")
    od ; return(emp) >,
  Ocnt: 0 >
Bye.
